nifi-nar-bundles/nifi-riemann-bundle/nifi-riemann-reporting-task/src/main/java/org/apache/nifi/reporting/riemann/RiemannReportingTask.java

```diff
@@ -0,0 +1,244 @@
+            .required(true).defaultValue("true").addValidator(StandardValidators.BOOLEAN_VALIDATOR).build();
+    private static final PropertyDescriptor SEND_NIFI_METRICS = new PropertyDescriptor.Builder().name("NiFi Metrics").description("Forwards aggregated data flow metrics to Riemann")
+            .allowableValues("true", "false").required(true).defaultValue("true").addValidator(StandardValidators.BOOLEAN_VALIDATOR).build();
+    private static final PropertyDescriptor SEND_PROCESSOR_METRICS = new PropertyDescriptor.Builder().name("Processor Metrics").description("Forwards metrics for individual processor to Riemann")
+            .allowableValues("true", "false").required(true).defaultValue("true").addValidator(StandardValidators.BOOLEAN_VALIDATOR).build();
+    private static final PropertyDescriptor SEND_BULLETIN_MESSAGES = new PropertyDescriptor.Builder().name("Bulletin Messages").description("Forwards messages from the Bulletin board to Riemann")
+            .allowableValues("true", "false").required(true).defaultValue("true").addValidator(StandardValidators.BOOLEAN_VALIDATOR).build();
+    private static final PropertyDescriptor MIN_BULLETIN_LEVEL = new PropertyDescriptor.Builder().name("Minimum Bulletin Level")
+            .description("Only forward bulletin messages at this level and above to Riemann").allowableValues(LogLevel.values()).required(true).defaultValue("WARNING").build();
+    protected Transport transport;
+    private volatile long lastObservedBulletinId = 0;
+    private volatile RiemannClient riemannClient = null;
+    private MetricsService metricsService;
+    private final VirtualMachineMetrics virtualMachineMetrics = VirtualMachineMetrics.getInstance();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = Lists.newArrayList();
+        properties.add(RIEMANN_HOST);
+        properties.add(RIEMANN_PORT);
+        properties.add(TRANSPORT_PROTOCOL);
+        properties.add(SERVICE_PREFIX);
+        properties.add(HOSTNAME);
+        properties.add(TAGS);
+        properties.add(SEND_JVM_METRICS);
+        properties.add(SEND_NIFI_METRICS);
+        properties.add(SEND_PROCESSOR_METRICS);
+        properties.add(SEND_BULLETIN_MESSAGES);
+        properties.add(MIN_BULLETIN_LEVEL);
+        properties.add(WRITE_TIMEOUT);
+        return Collections.unmodifiableList(properties);
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder().name(propertyDescriptorName).required(false).addValidator(StandardValidators.NON_EMPTY_VALIDATOR).expressionLanguageSupported(false).dynamic(true)
+                .build();
+    }
+
+    @OnStopped
+    public final void cleanUpClient() {
+        if (riemannClient != null) {
+            this.riemannClient.close();
```

Comment:
What if this throws an exception?  Hopefully the framework will catch it and log an error and provide a bulletin.